Tests/test_Affy.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+
+from Bio.Affy import CelFile
+import struct
+from numpy import array
+import numpy.testing
+
+class AffyTest(unittest.TestCase):
+    def setUp(self):
+        self.affy3 = "Affy/affy_v3_example.CEL"
+        self.affy4 = "Affy/affy_v4_example.CEL"
+    def tearDown(self):
+        pass
+    
+    # tests if the new strict mode complains about passing a non-file
+    def testAffyStrict(self):
+        with self.assertRaises(IOError) as context:
+            with open(self.affy3, "rb") as f:
+                record = CelFile.read("hello", strict=True)
+
+    # tests if the new strict mode is backwards compatible
+    def testAffyStrict(self):
+        with open(self.affy3, "rb") as f:
+            record = CelFile.read("hello")
+        assert record.DatHeader is None
```

Comment:
You open a handle to the affy3 file, but then don't use it. It looks like a mistake.